Holesky Launch

thedevbirb opened the pull request:
- Closes #159 
- Closes #295 
- Closes #308
- Related #236 


~TODO:~
- [x] Review section on Commit-Boost
- [x] Review section on collateral required
- [x] Provide updated `docker-compose.yml` w/ also metrics support
- [x] Make sure `cb-bolt-config.toml` is updated
- [x] Clean-up `testnets/holesky` folder

thedevbirb committed:
chore(testnet): holesky launch folder stub

thedevbirb committed:
chore(testnet): holesky launch folder stub

thedevbirb committed:
git(sidecar): add bolt-sidecar binaries to .gitignore

thedevbirb committed:
git(sidecar): add bolt-sidecar binaries to .gitignore

thedevbirb committed:
docker(sidecar): remove redundant dependencies during build

thedevbirb committed:
docker(sidecar): remove redundant dependencies during build

thedevbirb committed:
docs(holesky): README launch wip

update

thedevbirb committed:
docs(holesky): README launch wip

update

thedevbirb committed:
chore!(sidecar): rename env for telemetry options

thedevbirb committed:
chore!(sidecar): rename env for telemetry options

thedevbirb committed:
chore(sidecar): update .env.example

thedevbirb committed:
chore(sidecar): update .env.example

thedevbirb committed:
chore(sidecar): update Config.example.toml

thedevbirb committed:
chore(sidecar): update Config.example.toml

thedevbirb committed:
chore!(sidecar): remove short options from telemetry config

thedevbirb committed:
chore!(sidecar): remove short options from telemetry config

thedevbirb committed:
chore(holesky): add .gitignore

thedevbirb committed:
chore(holesky): add .gitignore

thedevbirb committed:
chore(mev-boost): add .env.example

thedevbirb committed:
chore(mev-boost): add .env.example

thedevbirb committed:
chore(mev-boost): update .gitignore

thedevbirb committed:
chore(mev-boost): update .gitignore

thedevbirb committed:
chore(holesky): update docker compose setup

thedevbirb committed:
chore(holesky): update docker compose setup

thedevbirb committed:
chore(holesky): move commit-boost config in separate directory

thedevbirb committed:
chore(holesky): move commit-boost config in separate directory

mempirate commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,791 @@
+
+**A synced Geth client:**
+
+At the moment Bolt is only compatible with the Geth execution client. Running
```
Clarify that you just need one synced Geth client (with the correct namespaces enabled!). This doesn't mean that it doesn't work with beacon nodes connected to other EL clients

mempirate commented on testnets/holesky/README.md:
````diff
@@ -0,0 +1,791 @@
+running:
+
+```bash
+cd testnets/holesky && docker compose build
````
Would add `docker compose up -d` here

mempirate commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,791 @@
+
+This will run all modules in Docker containers.
+
+> [!IMPORTANT] > `bolt-boost` will be exposed at `pbs.port` (18551 by default, set with `BOLT_SIDECAR_BUILDER_PROXY_PORT`), and your beacon node MUST be configured
```
This admonition is incorrectly formatted

mempirate commented on testnets/holesky/README.md:
````diff
@@ -0,0 +1,791 @@
+./bolt-sidecar-cli
+```
+
+## Docker
````
Let's call this Docker Mode, we can then do Native Mode and Commit-Boost Mode (just to make the naming a bit easier)

mempirate commented on testnets/holesky/README.md:
````diff
@@ -0,0 +1,791 @@
+```bash
+cd testnets/holesky && docker compose build
+```
+
````
Let's document observability here as well

thedevbirb replied:
Let me know if this is ok or if you would like it more detailed!

mempirate commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,791 @@
+Once you are successfully running the Bolt sidecar you need to register on-chain
+on the Bolt Registry to successfully receive preconfirmation requests from users
+and RPCs. This step is needed to prove that your commitments are economically
+backed.
```
"This step is needed to provide economic security to your commitments."

mempirate commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,791 @@
+active validators into the protocol, while the latter is used to manage to
+register as an operator into the system and integrate with the restaking
+protocols.
+
```
Let's add supported collaterals and their addresses per restaking protocol. We just support the eth derivatives for now.

mempirate commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,791 @@
+
+Until the Pectra hard-fork will be activated, the contract will also expose a `registerValidatorUnsafe` function
+that will not check the BLS signature. This is gated by a feature flag that will be turned off post-Pectra and
+will allow us to test the registration flow in a controlled environment.
```
I think we need some additional information here on how to register these validators. Can work on some scripts like last time.

thedevbirb replied:
If that's okay with you I'll update this section once we have the scripts!

mempirate reviewed (changes requested):
Great work!

thedevbirb committed:
chore(sidecar): more precise docs for config

thedevbirb committed:
chore(sidecar): more precise docs for config

merklefruit committed:
wip: pbs configs

merklefruit committed:
wip: pbs configs

merklefruit committed:
feat: working pbs docker compose

merklefruit committed:
feat: working pbs docker compose

thedevbirb committed:
fix(delegations-cli): non-descriptive error message

thedevbirb committed:
fix(delegations-cli): non-descriptive error message

thedevbirb committed:
fix(holesky): add bolt prefix to all services and containers to avoid conflicts

thedevbirb committed:
fix(holesky): add bolt prefix to all services and containers to avoid conflicts

Commit:
docs(holesky): update instructions

Commit:
docs(holesky): update instructions

thedevbirb committed:
fix(holesky): docker-compose sidecar entrypoint

thedevbirb committed:
fix(holesky): docker-compose sidecar entrypoint

thedevbirb committed:
fix(sidecar): .env.example

thedevbirb committed:
fix(sidecar): .env.example

Commit:
docs(holesky): smol chagnes

Commit:
docs(holesky): smol chagnes

thedevbirb committed:
chore(holesky): add simple bash script to create .env stub

thedevbirb committed:
chore(holesky): add simple bash script to create .env stub

Commit:
docs(holesky): smol changes

Commit:
docs(holesky): smol changes

Commit:
docs(holesky): validator registration script

Commit:
docs(holesky): validator registration script

Commit:
docs(holesky): operator registration

Commit:
docs(holesky): operator registration

thedevbirb committed:
chore(sidecar): update Config.example.toml

thedevbirb committed:
chore(sidecar): update Config.example.toml

thedevbirb committed:
chore(sidecar): review CLI options and their defaults

thedevbirb committed:
chore(sidecar): review CLI options and their defaults

thedevbirb committed:
chore(sidecar): add Config.toml to .gitignore

thedevbirb committed:
chore(sidecar): add Config.toml to .gitignore

thedevbirb committed:
feat(sidecar): allow reading TOML config from default path

thedevbirb committed:
feat(sidecar): allow reading TOML config from default path

thedevbirb committed:
chore(holesky): add *.toml to .gitignore

thedevbirb committed:
chore(holesky): add *.toml to .gitignore

thedevbirb committed:
fix(holesky): broken links in README, use TOML config file instead of .env for the sidecar

thedevbirb committed:
fix(holesky): broken links in README, use TOML config file instead of .env for the sidecar

thedevbirb committed:
chore(holesky): update docker-compose.yml

thedevbirb committed:
chore(holesky): update docker-compose.yml

thedevbirb committed:
fix(sidecar): min_priority_fee is now u128 and not NonZero<u128>

This caused issue when parsing from TOML string. Also if that value is
zero is not a problem actually

thedevbirb committed:
fix(sidecar): min_priority_fee is now u128 and not NonZero<u128>

This caused issue when parsing from TOML string. Also if that value is
zero is not a problem actually

thedevbirb committed:
fix(holesky): cat -> cp in README

thedevbirb committed:
fix(holesky): cat -> cp in README

thedevbirb committed:
chore(holesky): delegations flag in README

thedevbirb committed:
chore(holesky): delegations flag in README

thedevbirb committed:
fix(holesky): provide delegations cli volume

thedevbirb committed:
fix(holesky): provide delegations cli volume

thedevbirb committed:
chore(holesky): README fmt

thedevbirb committed:
chore(holesky): README fmt

thedevbirb committed:
chore(holesky): validator controller reference on README

thedevbirb committed:
chore(holesky): validator controller reference on README

thedevbirb committed:
fix(holesky): stale flags in README

thedevbirb committed:
fix(holesky): stale flags in README

thedevbirb committed:
chore(sidecar): mark required fields in the Config.example.toml

thedevbirb committed:
chore(sidecar): mark required fields in the Config.example.toml

thedevbirb committed:
fix(holesky): wrong code links and snippets in README

thedevbirb committed:
fix(holesky): wrong code links and snippets in README

thedevbirb committed:
fix(contracts): move operators scripts into new operators folder

thedevbirb committed:
fix(contracts): move operators scripts into new operators folder

thedevbirb committed:
feat(holesky): EL script for StrategyManager.depositIntoStrategy

thedevbirb committed:
feat(holesky): EL script for StrategyManager.depositIntoStrategy

thedevbirb committed:
feat(holesky): update scripts + README for Symbiotic integration

thedevbirb committed:
feat(holesky): update scripts + README for Symbiotic integration

thedevbirb committed:
feat(holesky): test commit grafana dashboard and prometheus setup

thedevbirb committed:
feat(holesky): test commit grafana dashboard and prometheus setup

thedevbirb committed:
fix(holesky): formatting in README

thedevbirb committed:
fix(holesky): formatting in README

thedevbirb committed:
chore(holesky): better explanation of operator config steps for EL

thedevbirb committed:
chore(holesky): better explanation of operator config steps for EL

thedevbirb committed:
fix(holesky): config options for Symbiotic guide

Don't use operators.json file but provide the RPC URL using an
enviroment value

thedevbirb committed:
fix(holesky): config options for Symbiotic guide

Don't use operators.json file but provide the RPC URL using an
enviroment value

thedevbirb committed:
chore(holesky): upgrade vault addresses for Symbiotic

thedevbirb committed:
chore(holesky): upgrade vault addresses for Symbiotic

Commit:
fix(contracts/scripts): register Symbiotic operator script

Commit:
fix(contracts/scripts): register Symbiotic operator script

merklefruit committed:
chore(holesky): updated bolt-cli usage guide

merklefruit committed:
chore(holesky): updated bolt-cli usage guide

merklefruit committed:
fix: broken links

merklefruit committed:
fix: broken links

thedevbirb committed:
chore!(sidecar,holesky): drop support for TOML, use .env everywhere

thedevbirb committed:
chore!(sidecar,holesky): drop support for TOML, use .env everywhere

thedevbirb committed:
fix(holesky): docker compose setup -- grafana, prometheus

thedevbirb committed:
fix(holesky): docker compose setup -- grafana, prometheus

thedevbirb committed:
chore(holesky): update README after dropping TOML support

thedevbirb committed:
chore(holesky): update README after dropping TOML support

thedevbirb committed:
Update testnets/holesky/README.md

Co-authored-by: nicolas <48695862+merklefruit@users.noreply.github.com>

thedevbirb committed:
Update testnets/holesky/README.md

Co-authored-by: nicolas <48695862+merklefruit@users.noreply.github.com>

thedevbirb committed:
chore(holesky): update rc image for Helix

thedevbirb committed:
chore(holesky): update rc image for Helix

thedevbirb committed:
chore(holesky): update rc image for Helix

thedevbirb committed:
fix(holesky): remove toml volume from dockerfile

thedevbirb committed:
fix(holesky): remove toml volume from dockerfile

thedevbirb committed:
fix(holesky): remove toml volume from dockerfile

thedevbirb committed:
feat(holesky): cadvisor dashboard stub

thedevbirb committed:
feat(holesky): cadvisor dashboard stub

thedevbirb committed:
feat(holesky): cadvisor dashboard stub

thedevbirb committed:
fix(sidecar): test/CI

thedevbirb committed:
fix(sidecar): test/CI

thedevbirb committed:
fix(sidecar): test/CI

thedevbirb committed:
fix(holesky): restore commit-boost observability previosly removed by mistake

thedevbirb committed:
fix(holesky): restore commit-boost observability previosly removed by mistake

thedevbirb committed:
fix(holesky): restore commit-boost observability previosly removed by mistake

Commit:
fix(holesky/docs): broken links

Commit:
fix(holesky/docs): broken links

Commit:
fix(holesky/docs): broken links

thedevbirb committed:
fix(holesky): cAdvisor only scans for the services in the current docker compose setup

thedevbirb committed:
fix(holesky): cAdvisor only scans for the services in the current docker compose setup

Commit:
feat(holesky): update networking & default values

Commit:
feat(holesky): update networking & default values

Commit:
Merge branch 'lore/feat/holesky-launch' of github.com:chainbound/bolt into lore/feat/holesky-launch

Commit:
feat(holesky): update networking & default values

Commit:
fix(holesky): fix conflict

Commit:
fix(holesky): add custom network to compose

Commit:
fix(holesky): add custom network to compose

Commit:
fix(holesky): typo

Commit:
fix(holesky): typo

Commit:
fix(holesky): genesis fork version on builder

Commit:
fix(holesky): genesis fork version on builder

Commit:
fix(holesky): cb port description

Commit:
fix(holesky): cb port description

Commit:
docs(holesky): document cb status

Commit:
docs(holesky): document cb status

merklefruit commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,1096 @@
+releases page](https://github.com/Commit-Boost/commit-boost-client/releases)
+
+A commit-boost configuration file with Bolt support is provided at
+[`cb-bolt-config.toml`](./cb-bolt-config.toml). This file has support for the
```
broken link

merklefruit commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,1096 @@
+A commit-boost configuration file with Bolt support is provided at
+[`cb-bolt-config.toml`](./cb-bolt-config.toml). This file has support for the
+custom PBS module ([bolt-boost](../../bolt-boost)) that implements the
+[constraints-API](https://chainbound.github.io/bolt-docs/api/builder), as well
```
broken link

merklefruit commented on testnets/holesky/README.md:
````diff
@@ -0,0 +1,1096 @@
+- Clone the Bolt repo and navigate to the [contracts](https://github.com/chainbound/bolt/tree/unstable/bolt-contracts) directory:
+
+```bash
+git clone https://github.com/chainbound/bolt
````
probably a good idea to include the `--branch v0.3.0` flag here like we do in other places

merklefruit reviewed (changes requested):
smol nits

thedevbirb committed:
chore(holesky): re-add bolt-default network to all services

Commit:
feat(contracts): smol changes

Commit:
docs(holesky): smol changes

Commit:
fix: merge

Commit:
fix: merge conflict

thedevbirb committed:
chore: remove mev-boost folder again

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
fix(holesky/docs): broken links

thedevbirb committed:
chore(holesky/env): add BOLT_SIDECAR_ENABLE_UNSAFE_LOOKAHEAD in .env.example

Commit:
docs(holesky): clarify order of operations

Commit:
docs(holesky): callout fix

merklefruit committed:
wip(registry): new storage layout with indexes

thedevbirb committed:
fix(holesky/docks): broken link to contract

thedevbirb committed:
fix(holesky/docs): table of contents

Commit:
docs(holesky): updates

thedevbirb committed:
fix(holesky/docs): toc

merklefruit committed:
feat: use separate validators library

merklefruit committed:
feat: upgrade validators script

merklefruit committed:
chore: added more tests

merklefruit committed:
feat: updated libraries

thedevbirb committed:
Merge pull request #335 from chainbound/jonas/docs/updates

docs(holesky): clarify order of operations

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

merklefruit committed:
feat: update library versions

merklefruit committed:
chore: ci

merklefruit committed:
chore: upgrade manager script

Commit:
chore(contracts): forge fmt

Commit:
feat(contracts): update tests

Commit:
docs(contracts): update deployments

Commit:
chore(contracts): TODO comment

Commit:
Merge pull request #336 from chainbound/nico/feat/cheaper-unsafe-optin

feat(registry): cheaper on-chain batch opt-in (48k gas)

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
chore(holesky/docker): update bolt-mev-boost image

thedevbirb committed:
chore(holesky): tag rc2 for bolt-sidecar

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
chore(sidecar): error log

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

Commit:
feat(cli): basefee + priority fee flags

Commit:
Merge branch 'lore/feat/holesky-launch' of github.com:chainbound/bolt into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

mempirate commented on testnets/holesky/README.md:
```diff
@@ -0,0 +1,1112 @@
+First you'll need to deposit some collateral in the form of whitelisted ETH derivative tokens that need to
+be restaked in either the Symbiotic or EigenLayer restaking protocols. Bolt is compatible with the following ETH derivative tokens on Holesky:
+
+- [Symbiotic Vaults](https://docs.symbiotic.fi/deployments#vaults)
```
Link should be removed here (we don't use their vaults)

merklefruit replied:
for now it was updated to the correct version

Commit:
wip: logs

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
chore(sidecar): remove extra logs

merklefruit committed:
chore: update send command impl

merklefruit committed:
chore: cleanup libraries, rm unused code

merklefruit committed:
chore: fmt

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
chore(holesky/docker): mount volumes on predefined paths

thedevbirb committed:
chore(holesky): remove defaults delegations.json file, no more needed

thedevbirb committed:
chore(holesky/docs): update docs reflecting docker compose changes; remove last references to deprecated TOML config file

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
fix(holesky/docs): broken symbiotic docs link

thedevbirb committed:
Merge branch 'unstable' into lore/feat/holesky-launch

thedevbirb committed:
fix(holesky): mev-boost.env.example can lead to a wrong starting config

merklefruit committed:
chore(sidecar): small nits

merklefruit committed:
chore(sidecar): small nits

merklefruit committed:
feat(sidecar): robust generic future retry system

merklefruit committed:
feat: rm validator indexes in favor of pubkeys

merklefruit committed:
chore(sidecar): add docs

merklefruit committed:
chore(sidecar): rm primitives todo comment

merklefruit committed:
chore(sidecar): rm state fetcher todo

merklefruit committed:
chore(ci): ignore eigenlayer docs website

merklefruit committed:
chore: refactor sidecar modules and docs

merklefruit committed:
chore: more docs + refactor

thedevbirb committed:
feat(bolt-cli): add Dockerfile

thedevbirb committed:
Merge pull request #362 from chainbound/nico/chore/refactor

chore(sidecar): minimal module refactor + add missing docs

thedevbirb committed:
chore(holesky/pbs): update helix config

thedevbirb committed:
Merge pull request #357 from chainbound/lore/fix/docker-volumes

chore(holesky/docker): mount volumes on predefined paths

merklefruit committed:
Merge pull request #364 from chainbound/lore/chore/update-helix-config

chore(holesky/pbs): update helix config

merklefruit committed:
fix: rm validator indexes in cb config file

thedevbirb committed:
Merge pull request #361 from chainbound/nico/feat/solve-outstanding-todos

feat: solve outstanding TODOs

thedevbirb committed:
Merge pull request #355 from chainbound/chore/cleanup-libraries

chore: cleanup libraries, rm unused code

merklefruit committed:
fix: broken link in md doc

thedevbirb committed:
fix!(config): remove empty environment variables set like 'VAR=' when parsing options

thedevbirb committed:
Merge pull request #367 from chainbound/lore/fix/config

fix!(config): remove empty environment variables set like 'VAR=' when parsing options

merklefruit committed:
fix: make cargo chef detect binary

thedevbirb committed:
Merge pull request #363 from chainbound/lore/feat/bolt-cli-docker

Add Dockerfile to `bolt` CLI

thedevbirb committed:
fix(holesky/docker): add environment variable replacement in entrypoint and not in 'environment'

thedevbirb committed:
chore(devnet): bump helix image

thedevbirb committed:
fix(sidecar): init fmt at the absolute beginning

merklefruit committed:
chore(sidecar): cleanup + remove TODOs (continue)

thedevbirb committed:
fix(sidecar/config): missing field in Debug impl

Commit:
fix(sidecar): don't remove constraints before accounting

thedevbirb committed:
chore: restore tracing in telemetry stack init

thedevbirb committed:
fix(holesky/docs): docker compose command

thedevbirb committed:
Merge pull request #371 from chainbound/lore/fix/docker-again

fix(holesky/docker): add environment variable replacement in entrypoint

merklefruit commented:
Reminder: when merging this branch, we must not delete it immediately, 
because people are relying on it for links to the guide.

merklefruit committed:
chore: eyre::Result

thedevbirb committed:
Merge pull request #370 from chainbound/nico/chore/sidecar-cleanup

chore(sidecar): cleanup + remove TODOs (continue)

merklefruit committed:
fix: revert helix image tag to local

merklefruit committed:
chore: use alloy::hex::encode_prefixed where possible

thedevbirb committed:
fix(sidecar): remove stale templates in case of missed slots

thedevbirb committed:
Merge pull request #374 from chainbound/nico/chore/fixes

small fixes

merklefruit committed:
Merge pull request #375 from chainbound/lore/fix/remove-stale-templates

Remove stale templates in case of missed slots

thedevbirb committed:
chore(holesky): bump helix image, add helix_website=debug log directive

thedevbirb commented:
🚀
